internal/querycoord/impl.go

```diff
@@ -88,6 +88,63 @@ func (qc *QueryCoord) GetStatisticsChannel(ctx context.Context) (*milvuspb.Strin
 	}, nil
 }
 
+func (qc *QueryCoord) checkAnyReplicaAvailable(collectionID UniqueID) bool {
+	replicas, _ := qc.meta.getReplicasByCollectionID(collectionID)
+	// Note: copy from GetShardLeaders
+	numShards := 2
+	if len(replicas) > 0 {
+		numShards = len(replicas[0].ShardReplicas)
+	}
```

Comment:
NIT: Feels weird that it has to have replicas to get shard number right. Maybe shard number could be retrieved from collection meta.